File: selectionSort.js

function selectionSort(array) {
    // Write your code here.
    let i = 0
    while (i < array.length - 1) {
        let j = i;
        for (let k = i + 1; k < array.length; k++) {

            console.log(k)
            if (array[j] > array[k]) { j = k }

        }
        swap(i, j, array)

        i++;

    }
    return array;
}
function swap(i, j, array) {
    const temp = array[j];
    array[j] = array[i];
    array[i] = temp
}
const y = selectionSort([1, 4, 2, 3])
console.log(y)

File: selectionSort.py
def selectionSort(array):
    i = 0
    while i < len(array) - 1:
        j = i
        for k in range(i+1, len(array)):
            if (array[j] > array[k]):
                j = k
        swap(i, j, array)
        i += 1
    return array


def swap(i, j, array):
    array[i], array[j] = array[j], array[i]


y = selectionSort([1, 4, 2, 3])
print(y)
